pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -53,14 +53,14 @@ func (e *ErrorWithCodes) Error() string {
 
 var (
 	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|InvalidAuthenticationTokenTenant|SignatureDoesNotMatch|Authentication failed|AuthFailure|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId)`)
-	quotaExceededRegexp                 = regexp.MustCompile(`(?i)((?:^|[^t]|(?:[^s]|^)t|(?:[^e]|^)st|(?:[^u]|^)est|(?:[^q]|^)uest|(?:[^e]|^)quest|(?:[^r]|^)equest)LimitExceeded|Quotas|Quota exceeded|exceeded quota|Quota has been met|QUOTA_EXCEEDED|Maximum number of ports exceeded)`)
+	quotaExceededRegexp                 = regexp.MustCompile(`(?i)((?:^|[^t]|(?:[^s]|^)t|(?:[^e]|^)st|(?:[^u]|^)est|(?:[^q]|^)uest|(?:[^e]|^)quest|(?:[^r]|^)equest)LimitExceeded|Quotas|Quota exceeded|exceeded quota|Quota has been met|QUOTA_EXCEEDED|Maximum number of ports exceeded|does not have enough resources)`)
```

Comment:
How about using `ZONE_RESOURCE_POOL_EXHAUSTED_WITH_DETAILS` here instead of `does not have enough resources`? As far as I understand this is the error code returned by the provider - shouldn't we prefer using such error codes instead of error messages?

I think this entire mechanism already suffers from an abundance of messages that are frankly too generic, leading to wrong classification or to multiple classifications due to multiple regex matching. IMO we should aim to use the most unique string that identifies the error.

Reply:
Fair enough, I used the less specific part of the message to potentially catch more errors, but it comes at the price of having false positives, of course. I change it.